pegasus_wrapper/resource_request.py

```diff
@@ -116,6 +129,8 @@ def unify(self, other: ResourceRequest) -> ResourceRequest:
             job_time_in_minutes=other.job_time_in_minutes
             if other.job_time_in_minutes
             else self.job_time_in_minutes,
+            exclude_list=other.exclude_list,
+            run_on_single_node=other.run_on_single_node,
```

Comment:
It looks like both of these are missing a conditional check to pick from either `self` or `other`. 
```suggestion
            exclude_list=other.exclude_list if other.exclude_list else self.exclude_list,
            run_on_single_node=other.run_on_single_node if other.run_on_single_node else self.run_on_single_node,
```

Reply:
@hassanzadeh This comment still appears to be outstanding as an issue to resolve.